### PR TITLE
Validations and tests

### PR DESCRIPTION
**Pull request by IsmailChangezi**
## Description 
- Install Rspec 
- Validate methods
- Write tests

### CODE REVIEW

File: spec/post_spec.rb
Comment (fobadara) [inline spec/post_spec.rb:1]: [REQUIRED] Kindly include test for `recent_comments` as it is a requirement to test custom methods

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @IsmailChangezi, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Clean code 
✅ Dry code

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @IsmailChangezi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

 ## Required Changes ♻️
- [ ] When I run the tests I got an error kindly make sure to fix this so your app is working as expected

Screenshot_20230701_223641

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Whoistolu — APPROVED** [review]
### Hi @IsmailChangezi 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Great effort fixing the failing test cases 👍🏽 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [ ] [OPTIONAL] Kindly ensure that the PR has more description. It should be a summary of the changes that occurred on this milestone.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**